.editorconfig

```diff
@@ -0,0 +1 @@
dotnet_naming_symbols.private_or_internal_static_field.required_modifiers = static

dotnet_naming_symbols.types.applicable_kinds = class, struct, interface, enum
dotnet_naming_symbols.types.applicable_accessibilities = public, internal, private, protected, protected_internal, private_protected
dotnet_naming_symbols.types.required_modifiers =

dotnet_naming_symbols.non_field_members.applicable_kinds = property, event, method
dotnet_naming_symbols.non_field_members.applicable_accessibilities = public, internal, private, protected, protected_internal, private_protected
dotnet_naming_symbols.non_field_members.required_modifiers =

dotnet_naming_symbols.non_public_const_fields.applicable_kinds = field
dotnet_naming_symbols.non_public_const_fields.applicable_accessibilities = internal, private, protected, protected_internal, private_protected
dotnet_naming_symbols.non_public_const_fields.required_modifiers = const

dotnet_naming_symbols.non_public_static_readonly_fields.applicable_kinds = field
dotnet_naming_symbols.non_public_static_readonly_fields.applicable_accessibilities = private, protected, protected_internal, private_protected
dotnet_naming_symbols.non_public_static_readonly_fields.required_modifiers = readonly, static

dotnet_naming_symbols.method_parameters.applicable_kinds = parameter
dotnet_naming_symbols.method_parameters.applicable_accessibilities =
dotnet_naming_symbols.method_parameters.required_modifiers =

dotnet_naming_symbols.local.applicable_kinds = local
dotnet_naming_symbols.local.applicable_accessibilities = local
dotnet_naming_symbols.local.required_modifiers =

# naming styles

dotnet_naming_style.pascal_case.required_prefix =
dotnet_naming_style.pascal_case.required_suffix =
dotnet_naming_style.pascal_case.word_separator =
dotnet_naming_style.pascal_case.capitalization = pascal_case

dotnet_naming_style.begins_with_i.required_prefix = I
dotnet_naming_style.begins_with_i.required_suffix =
dotnet_naming_style.begins_with_i.word_separator =
dotnet_naming_style.begins_with_i.capitalization = pascal_case

dotnet_naming_style.begins_with_m_.required_prefix = m_
dotnet_naming_style.begins_with_m_.required_suffix =
dotnet_naming_style.begins_with_m_.word_separator =
dotnet_naming_style.begins_with_m_.capitalization = pascal_case

dotnet_naming_style.begins_with_s_.required_prefix = s_
dotnet_naming_style.begins_with_s_.required_suffix =
dotnet_naming_style.begins_with_s_.word_separator =
dotnet_naming_style.begins_with_s_.capitalization = pascal_case

dotnet_naming_style.begins_with_k_.required_prefix = k_
dotnet_naming_style.begins_with_k_.required_suffix =
dotnet_naming_style.begins_with_k_.word_separator =
dotnet_naming_style.begins_with_k_.capitalization = pascal_case

dotnet_naming_style.camel_case.required_prefix =
dotnet_naming_style.camel_case.required_suffix =
dotnet_naming_style.camel_case.word_separator =
dotnet_naming_style.camel_case.capitalization = camel_case

```

Comment:
I se a lot of  `^M` characters in this file, is this, are these expected? What are they?

Reply:
> The ^M is a carriage-return character. If you see this, you're probably looking at a file that originated in the DOS/Windows world, where an end-of-line is marked by a carriage return/newline pair, whereas in the Unix world, end-of-line is marked by a single newline.

Per this [thread](https://unix.stackexchange.com/questions/32001/what-is-m-and-how-do-i-get-rid-of-it).

Ah, it seems like that file was originally pushed up with enforced Windows line endings (grabbed from Boss Room repo).
I've modified line endings on this file, so that a line ending is just a newline.
I did this locally, but I'll create a task to apply this to our repository's git settings!